enoslib/tests/unit/test_network_utils.py

```diff
@@ -245,6 +245,47 @@ def test_expansion_no_symetric(self):
             self.assertEquals('20mbit', d['rate'])
             self.assertEquals('20ms', d['delay'])
 
+    def test_same_src_and_dest_defaults_embedded(self):
+        constraints = [{
+            'src': 'grp1',
+            'dst': 'grp1',
+            'rate': '20mbit',
+            'delay': '20ms'
+            }]
+        network_constraints = {
+            'default_rate': '10mbit',
+            'default_delay': '10ms',
+            'constraints' : constraints
+        }
+        descs = _generate_actual_grp_constraints(network_constraints)
+
+        self.assertEquals(1, len(descs))
+        self.assertDictEqual(constraints[0], descs[0])
+        for d in descs:
+            self.assertTrue('grp1' == d['src'])
+            self.assertTrue('grp1' == d['dst'])
+
+    def test_same_src_and_dest_without_defaults(self):
+        roles = {
+            'grp1': [Host('node1')],
+            'grp2': [Host('node2')],
+            'grp3': [Host('node3')]
+        }
+        constraints = [{
+            'src': 'grp1',
+            'dst': 'grp1'
+        }]
+        network_constraints = {
+            'default_rate': '10mbit',
+            'default_delay': '10ms',
+            'constraints': constraints
+        }
+        descs = _build_grp_constraints(roles, network_constraints)
```

Comment:
Could you test the len of descs ? should be 1.

Reply:
I read to quickly :)
I just want to make sure than the group with `src==dst==1` appears only once.